### DISCUSSION TITLE
forrtl: error (65): floating invalid

### DISCUSSION THREAD

**Discussion by baitao225** [NCAR/hrldas#113] [Q&A] [answered]
It reported an error as follows in the./hrldas.exe namelist.hrldas step while hrldas running a global simulation:
![image](https://github.com/NCAR/hrldas/assets/146513997/1f03034e-8b2a-4518-834c-d8ab1746bccc)

I found the position of 'SNOW HEIGHT NOT FOUND - the VALUE DEFINED IN LSMINIT ' in the code, namely noahmp/drivers/hrldas/NoahmpInitMainMod. F90 line 51. In combination with the context, we can see that the model makes snow height equal to swe*0.05, that is, NoahmpIO%SNOWH(I,J) = NoahmpIO%SNOW(I,J) * 0.005.

My snow data ranges from 0 to 88652.9, and the snow data greater than 2000 has been corrected in NoahmpInitMainMod.F90. So I don't know why I get forrtl: error (65): floating invalid.  Even if I manually correct the snow data greater than 2000, I still get this error.
I'd appreciate it if you'd like to help me to deal with this problem.

**Comment by cenlinhe** [comment]
Based on your error message, please check Line 123 of your NoahmpInitMainMod.f90 (not .F90) to see the place where the model crashed.
If you are using ERA5 data to drive the model, your soil thickness in namelist.hrldas needs to be set to the same values as ERA5 instead of the default values. Please double check your soil thickness value in your HRLDAS_setup_file to see if it is consistent with your setup in the namelist.

**Reply by baitao225** [reply]
Sorry I didn't mention that I was using GLDAS data to drive the model. I have changed the soil thickness value in namelis.hrldas to be consistent with HRLDAS_setup_file, i.e. 0.1,0.3,0.7,1.5. But it still reported this error.
![image](https://github.com/NCAR/hrldas/assets/146513997/c7037ca1-9663-474c-8fc7-50510bfaf309)

I checked Line 123 of my NoahmpInitMainMod.f90 and found that there was a calculation here.
![image](https://github.com/NCAR/hrldas/assets/146513997/6a97f63b-b18f-4bdc-90b4-c7b310f5d9b1)

I have checked that the TSLB variable does not have a 0 value in the HRLDAS_setup_file. Considering the forrtl: error (65): floating invalid, could it be because the result was too small？
How should I solve this problem?

**Reply by tslin2** [reply]
you can try printing out these variables before this equation, and to see which variables have invalid, missing (i.e., NaN), or strange values.
For example, add the following code in source code (.F90) before this equation and recompile and rerun the model.

print *, 'debug:',  HLICE, GRAV0, PSISAT, NoahmpIO%TSLB(I,NS,J), T0, BEXP, SMCMAX

you should see these values out while running the simulation!

**Reply by cenlinhe** [reply]
I agree with @tslin2 Typically, this error occurs because you have some unreasonable values in your initial condition file

**Reply by NisKarki** [reply]
Hi @baitao225 , I am running in the same exact error. Were you able to resolve it?

I printed out the variables as suggested by @tslin2 and it was the TSLB variable. Apparently it was reading in no data value of  -9.9999996E+35 resulting in forrtl: error (65): floating invalid. Can you confirm if the nodata handling is what's causing this @cenlinhe?

**Reply by NisKarki** [reply]
I downloaded the hrldas setup file from the drive that you shared. Here is the difference in TSLB variable from create_forcing I got (top) and from the drive (bot). Could it be because of this although I guess its the result of different EXPAND_LOOPS values?
![my_hrldas_setup](https://github.com/NCAR/hrldas/assets/133700617/b9aa7692-3351-4f37-815f-35a7d689564d)
![from_drive](https://github.com/NCAR/hrldas/assets/133700617/e39ce5c1-14df-47a2-bea8-38175bfd51fd)

**Reply by NisKarki** [reply, marked as answer]
For anyone facing the same issue, I increased the EXPAND_LOOP value to 10 and it did solve the issue for soil initialization equation but got same error for groundwater init module. After further increasing to 15, the model ran without errors. However, I don't know whether it has any impact in the actual simulation results @cenlinhe.

**Reply by cenlinhe** [reply]
You are doing correctly by increasing EXPAND_LOOP. A larger EXPAND_LOOP value is needed to resolve coastal points and small island points. As long as those land points are resolved by a large EXPAND_LOOP and the model runs successfully, it won't affect the results if you further increasing the EXPAND_LOOP values.
